ledger-core/virtual/execute/execute.go

```diff
@@ -678,7 +680,11 @@ func (s *SMExecute) stepSendOutgoing(ctx smachine.ExecutionContext) smachine.Sta
 			return ctx.Error(throw.E("failed to publish bargeInCallback"))
 		}
 	} else {
-		s.outgoing.CallRequestFlags = payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall)
+		if s.outgoingSentCounter < MaxOutgoingSendCount {
```

Comment:
It can be written better like:
```suggestion
		if s.outgoingSentCounter >= MaxOutgoingSendCount {
		     return ctx.Error(throw.E("outgoing retries limit"))
		}
		
		s.outgoing.CallRequestFlags = payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall)
```